### ISSUE TITLE
Template autoformatting at buildtime

### ISSUE THREAD

**Issue by nedtwigg** [casid/jte#209]
I hoped to add a `jte` section to [Spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle#table-of-contents).

But it looks like the formatter is tied pretty intimately to IntelliJ

- https://github.com/casid/jte-intellij/tree/master/src/main/java/org/jusecase/jte/intellij/language/format

It seems like it would be possible to make a pretty-printer by implementing a [`TemplateParserVisitor`](https://github.com/casid/jte/blob/main/jte/src/main/java/gg/jte/compiler/TemplateParserVisitor.java) which implements all the callbacks by appending to a big String.

Any thoughts?

**Comment by casid** [comment]
@nedtwigg in theorie yes, you could give it a try. For jte parts it should work, but I'm unsure how to format the underlying language (for instance HTML/JS/CSS). The `TemplateParser` does omit some stuff if configured, so I think the `gg.jte.TemplateConfig` would need to be set to `htmlCommentsPreserved = true` and `trimControlStructures = false`.

**Comment by edward3h** [comment]
If the template is being used in plain text mode it could be rendering anything. You can't generalise format in that case.

**Comment by nedtwigg** [comment]
True, but HTML is common enough that `TemplateConfig` has fields like `htmlCommentsPreserved`. Something can be useful without being universal.

**Comment by casid** [comment]
I don't have time to work on it, but will leave it open as @nedtwigg suggested.
